src/core/src/core_logic/PatchAssessor.py

```diff
@@ -91,3 +102,111 @@ def raise_if_telemetry_unsupported(self):
 
         self.composite_logger.log("{0}".format(Constants.TELEMETRY_COMPATIBLE_MSG))
 
+    # region - Auto-assessment extensions
+    def should_auto_assessment_run(self):
+        # get last start time
+        try:
+            assessment_state = self.read_assessment_state()
+            last_start_in_seconds_since_epoch = assessment_state['lastStartInSecondsSinceEpoch']
+        except Exception as error:
+            self.composite_logger.log_warning("No valid last start information available for auto-assessment.")
+            return True
+
+        # get minimum elapsed time required
+        min_elapsed_seconds_required = self.convert_iso8601_duration_to_total_seconds(Constants.MIN_AUTO_ASSESSMENT_INTERVAL)
+
+        # check if required duration has passed
+        elapsed_time_in_seconds = self.__get_seconds_since_epoch() - last_start_in_seconds_since_epoch
+        if elapsed_time_in_seconds < 0:
+            self.composite_logger.log_warning("Anomaly detected in system time now or during the last assessment run. Assessment will run anyway.")
+            return True
+        else:
+            return elapsed_time_in_seconds >= min_elapsed_seconds_required
+
+    def read_assessment_state(self):
+        """ Reads the assessment state file. """
+        self.composite_logger.log_debug("Reading assessment state...")
+        if not os.path.exists(self.assessment_state_file_path) or not os.path.isfile(self.assessment_state_file_path):
+            # Neutralizes directories
+            if os.path.isdir(self.assessment_state_file_path):
+                self.composite_logger.log_error("Assessment state file path returned a directory. Attempting to reset.")
+                shutil.rmtree(self.assessment_state_file_path)
+            # Writes a vanilla assessment statefile
+            self.write_assessment_state(first_write=True)
+
+        # Read (with retries for only IO Errors)
+        for i in range(0, Constants.MAX_FILE_OPERATION_RETRY_COUNT):
+            try:
+                with self.env_layer.file_system.open(self.assessment_state_file_path, mode="r") as file_handle:
+                    return json.load(file_handle)['assessmentState']
+            except Exception as error:
+                if i < Constants.MAX_FILE_OPERATION_RETRY_COUNT - 1:
+                    self.composite_logger.log_warning("Exception on assessment state read. [Exception={0}] [RetryCount={1}]".format(repr(error), str(i)))
+                    time.sleep(i + 1)
+                else:
+                    self.composite_logger.log_error("Unable to read assessment state file (retries exhausted). [Exception={0}]".format(repr(error)))
+                    raise
```

Comment:
This will fail the overall operation. Raises a question, how important is AssessmentState.json? Why the minimum time between auto assessment runs? Was this requirement raised and a known thing between our team and the Arc team?

Reply:
I see read_assessment_state() is currently only called in should_auto_assessment_run() with exception handling in there. So raising an exception here would  probably be fine, but the above comment still applies to write_assessment_state() since it's called within start_assessment(), and also in future cases where read_assessment may be called elsewhere